lib/backend/k8s/syncer.go

```diff
@@ -379,7 +408,20 @@ func (syn *kubeSyncer) performSnapshot() ([]model.KVPair, map[model.Key]bool, re
 
 		for _, c := range confList {
 			snap = append(snap, *c)
-			keys[c.Key] = true
+			keys[fmt.Sprintf("%s", c.Key)] = true
+		}
+
+		// Sync IP Pools.
+		poolList, err := syn.kc.List(model.IPPoolListOptions{})
+		if err != nil {
```

Comment:
What error are you expecting here that a 1s sleep and retry will be sufficient?  It's not clear from the code/comments.